### DISCUSSION TITLE
Self Host - Minimum Version Compatibility

### DISCUSSION THREAD

**Discussion by joshua5822** [getsentry/sentry-dotnet#2162] [Q&A] [answered]
I've started migrating from SharpRaven to using this package to send data to Sentry. Does the Self Hosted Sentry need to be at a minimum version to work with this project? The Sentries we have set up say 9.0.0 in the bottom left corner, so I don't think they have ever been updated. If I do have to get them updated, would the newer version still work with projects using SharpRaven without intervention. I am only updating one of the many projects that integrate to sentry.

**Comment by mattjohnsonpint** [comment, marked as answer]
You'll need Sentry version `20.6.0` or newer to use any Sentry .NET 3.x SDK, per [the migration doc](https://docs.sentry.io/platforms/dotnet/migration/#self-hosted-support).

... but also we have a [note in the changlog](https://github.com/getsentry/sentry-dotnet/blob/main/CHANGELOG.md#3170) indicating Sentry `21.9.0` for Sentry .NET version 3.17.0 or newer (or manually disabling client reports in your Sentry options).

... but additionally, we just released some new features that enable symbolication using .NET Portable PDB.  While not absolutely required - we think this is super valuable.  It does require Symbolicator 0.6.0 on the server, which I *think* gets updated during installation of Sentry.  There's a note about [component drift](https://github.com/getsentry/self-hosted/blob/master/CHANGELOG.md#21120) in the `21.12.0` Sentry changelog.


So long story short, YES you should update the Sentry self-hosted server.  Latest is best, but at minimum you'll need `20.6.0`.

If you want to take things one step at a time, you *might* try running the older `2.1.8` version of the Sentry .NET SDK against your older self-hosted Sentry.  Just keep in mind we only maintain the latest versions of each.
